test_runner/fixtures/neon_fixtures.py

```diff
@@ -753,20 +753,12 @@ def __exit__(
         # Stop all the nodes.
         if self.env:
             log.info("Cleaning up all storage and compute nodes")
-            self.env.endpoints.stop_all()
-            for sk in self.env.safekeepers:
-                sk.stop(immediate=True)
-
-            for pageserver in self.env.pageservers:
+            self.env.stop(
+                immediate=True,
                 # if the test threw an exception, don't check for errors
                 # as a failing assertion would cause the cleanup below to fail
-                if exc_type is not None:
-                    pageserver.assert_no_metric_errors()
-
-                pageserver.stop(immediate=True)
-
-            self.env.attachment_service.stop(immediate=True)
-
+                ps_assert_metric_no_errors=(exc_type is not None),
```

Comment:
I'm not sure I fully understand the logic behind this part, why we don't want to check `pageserver.log` for unexpected errors if there were no exceptions raised in the fixture?

Reply:
1. This is only a refactor, i.e., behavior is the same as before.
2. This doesn't check the logs, this checks the `/metrics`. The rationale is given in the comment, i.e., if something went wrong during the test, it might be that `/metrics` is unreachable, e.g., because pageserver is stopped.

Reply:
Right, I've confused metrics with logs.

Nevertheless, the logic seems to be inverted — `exc_type is not None` is true if an exception has happened (thus, we have exc_type and it's not None). The correct condition is

```suggestion
                ps_assert_metric_no_errors=(exc_type is None),
```

Seems, I've missed that in https://github.com/neondatabase/neon/pull/6295

Reply:
Hmm yeah, correct. Will let that slip into this PR